frontstage/views/sign_in/errors_routing.py

```diff
@@ -0,0 +1,77 @@
+import logging
+from abc import abstractmethod
+
+from typing import Dict
+
+from flask import render_template
+from structlog import wrap_logger
+
+from frontstage.notifications.notifications import AlertViaGovNotify
+
+logger = wrap_logger(logging.getLogger(__name__))
+
+
+class RoutingValidation:
+    """
+    This is the basic class that identify a generic routing type for any validation error that needs to be re routed.
+    All the abstract methods are mandatory to implement
+    """
+
+    def __init__(self, page: str, data: Dict):
+        """
+        :type page: page destination for the route function
+        :type data: it s dictionary of values to pass to the route function
+        """
+        self.page = page
+        self.data = data
+
+    @abstractmethod
+    def log_message(self, **log_parameters) -> str:
+        pass
```

Comment:
I wonder if we should instead have this (/ the concrete implementations) make use of the structlog, e.g.:

```python
def log_message(self, **log_parameters):
    logger.info('User account locked for too many attempts', **log_parameters)
```

i.e.: `.log_message(error=exc.oauth2_error)`

which would result in log rows like:
-> `{"event": "User account locked for too many attempts", "error": "User account locked", "level": "ERROR"...}`